src/System.Net.Ping/tests/FunctionalTests/UnixPingUtilityTests.cs

```diff
@@ -49,24 +49,48 @@ public class UnixPingUtilityTests
             int numPingsSent = ParseNumPingsSent(pingOutput);
             Assert.Equal(1, numPingsSent);
 
-            long rtt = UnixCommandLinePing.ParseRoundTripTime(pingOutput);
-            Assert.InRange(rtt, 0, long.MaxValue);
+            try
+            {
+                long rtt = UnixCommandLinePing.ParseRoundTripTime(pingOutput);
+                Assert.InRange(rtt, 0, long.MaxValue);
+            }
+            catch
+            {
+                Assert.Contains("should contain round trip time", pingOutput); // Will log the output
+                throw;
+            }
         }
 
         private static int ParseReturnedPacketSize(string pingOutput)
         {
-            int indexOfBytesFrom = pingOutput.IndexOf("bytes from");
-            int previousNewLine = pingOutput.LastIndexOf(Environment.NewLine, indexOfBytesFrom);
-            string number = pingOutput.Substring(previousNewLine + 1, indexOfBytesFrom - previousNewLine - 1);
-            return int.Parse(number);
+            try
+            {
+                int indexOfBytesFrom = pingOutput.IndexOf("bytes from");
+                int previousNewLine = pingOutput.LastIndexOf(Environment.NewLine, indexOfBytesFrom);
+                string number = pingOutput.Substring(previousNewLine + 1, indexOfBytesFrom - previousNewLine - 1);
+                return int.Parse(number);
+            }
+            catch
+            {
+                Assert.Contains("should contain packet size", pingOutput); // Will log the output
+                throw;
+            }
         }
 
         private static int ParseNumPingsSent(string pingOutput)
         {
-            int indexOfPacketsTransmitted = pingOutput.IndexOf("packets transmitted");
-            int previousNewLine = pingOutput.LastIndexOf(Environment.NewLine, indexOfPacketsTransmitted);
-            string number = pingOutput.Substring(previousNewLine + 1, indexOfPacketsTransmitted - previousNewLine - 1);
-            return int.Parse(number);
+            try
+            {
+                int indexOfPacketsTransmitted = pingOutput.IndexOf("packets transmitted");
+                int previousNewLine = pingOutput.LastIndexOf(Environment.NewLine, indexOfPacketsTransmitted);
+                string number = pingOutput.Substring(previousNewLine + 1, indexOfPacketsTransmitted - previousNewLine - 1);
+                return int.Parse(number);
+            }
+            catch
+            {
+                Assert.Contains("should contain num pings sent", pingOutput); // Will log the output
+                throw;
+            }
```

Comment:
Why add try/catches to both of these helpers rather than just extending up the try in the call site?